pkg/util/types.go

```diff
@@ -68,6 +68,21 @@ func AsSliceOfStrings(unk interface{}) ([]string, error) {
 	return result, nil
 }
 
+//AsMapOfStringsIntefaces attempts to convert unk to a map[string]interface{}
+func AsMapOfStringsIntefaces(unk interface{}) (map[string]interface{}, error) {
+	v := reflect.ValueOf(unk)
+	v = reflect.Indirect(v)
+
+	if v.Kind() != reflect.Map {
+		return make(map[string]interface{}), fmt.Errorf("cannot convert %v (%v) to map[string]interface{}", v.Type(), v)
```

Comment:
not returning nil? If one forgets to check the error and continues there is no difference to unk being an empty map of string to interface.